### PR TITLE
refs #14615 - properly map host id

### PR DESCRIPTION
**Pull request by thomasmckay**

### CODE REVIEW

File: lib/hammer_cli_import/contenthost.rb [hunk 1]
```diff
@@ -129,8 +129,8 @@ def post_import(_file)
         Dir.mkdir dir_name
         CSVHelper.csv_write_hashes(
           File.join(dir_name, "system-id_to_uuid-#{now.to_i}.map"),
-          [:system_id, :uuid, :org_id],
-          @map.sort_by { |x| [x[:org_id], x[:system_id], x[:uuid]] })
+          [:host_id, :uuid, :org_id],
```
Comment (ehelms) [inline lib/hammer_cli_import/contenthost.rb]: I thought the intent was system_id represented the Spacewalk/Sat 5 system_id and not the Foreman host_id ?

File: lib/hammer_cli_import/contenthost.rb [hunk 2]
```diff
@@ -93,7 +93,7 @@ def import_single_row(data)
         @map << {
           :org_id => data['organization_id'].to_i,
           :host_id => data['server_id'].to_i,
-          :uuid => c_host['uuid']}
+          :uuid => c_host['id']}
```
Comment (ehelms) [inline lib/hammer_cli_import/contenthost.rb]: Won't this still need to be the UUID in Candlepin in order for the transitioning systems to be able to register back? Trying to recall my understanding of how this works, but I thought it created a host object in Foreman with a subscription facet thus giving it a Candlepin UUID. On the real system, it de-registers from Sat 5, and then registers to Foreman passing along the Candlepin UUID to link the system to the pre-created host record.

Comment (thomasmckay) [inline lib/hammer_cli_import/contenthost.rb]: You're probably right. I will put candlepin uuid in this map file.

**Comment by thomasmckay** [comment]
```
$ tar -Oxvf  ~/tmp/SOURCES/*.tar.gz *.map
system-profile-transition-thomasmckay.usersys.redhat.com-1469020197-0.0.1/system-id_to_uuid-1469020197.map
system_id,uuid,org_id
210019988,3c96c53b-27f9-4ace-b2e2-22b136bb12ad,1
$ cat ~/.transition_data/systems-2016-07-20T13:09:57+0000.csv 
sat5,sat6,delete
210019988,28,
```

How does this look?

**Comment by ehelms** [comment]
Looks more correct to me (based on my limited knowledge of how it should work)

**Comment by ehelms** [comment]
In order to get the --delete to also work, I had to make the following changes as I think the change to use the hosts API only partially converted all of the persistent entity mappings so that in some places it still looked for `:systems` and others `:hosts`.

https://github.com/thomasmckay/hammer-cli-import/compare/14615-moar...ehelms:systems-to-hosts?expand=1

**Comment by lpramuk** [comment]
Tested changes. We are OK now. This is how it used to look like:
```
system_id,uuid,org_id
1000010009,40655cbc-8f16-48ca-9f81-69d35e6c1600,1
1000010010,fe8dfcc0-4701-4762-a989-8fc2defcc234,2
1000010011,c5011b37-ffbb-47ac-b00b-759162dd8a59,3
```

**Comment by lpramuk** [comment]
sat5 side is happy too:
```
# spacewalk-api --server=localhost --nologin system.transitionDataForSystem "$(</etc/sysconfig/rhn/systemid)"
$result = {
            'system_id' => '1000010011',
            'timestamp' => '1469032000',
            'hostname' => '<SAT6_FQDN>',
            'uuid' => 'e9ef086b-1a83-45db-ad52-1f89e76c742d'
          };
```

**Pull request merged**